subprojects/internal-testing/src/main/groovy/org/gradle/integtests/fixtures/TestClassExecutionResult.java

```diff
@@ -37,6 +38,31 @@ public interface TestClassExecutionResult {
 
     int getTestCount();
 
+    /**
+     * Asserts that the given tests have the given outcome for the given test class.
+     *
+     * In case of FAILED only one error message is assumed.
+     */
+    @SuppressWarnings("unchecked")
+    default TestClassExecutionResult assertTestOutcomes(TestOutcome status, String... testNames) {
+        if (status == TestOutcome.SKIPPED) {
+            return assertTestsSkipped(testNames);
+        }
+        for (String testName : testNames) {
+            switch (status) {
+                case PASSED:
+                    assertTestPassed(testName);
+                    break;
+                case FAILED:
+                    assertTestFailed(testName, new IsAnything<>());
```

Comment:
@octylFractal Here is interesting case, if amount of matchers is doesn't equal to amount of messages, then `assertTestFailed` fails. So assuming that at least one message will be presented. So this is becomes a limitation of `assertTestOutcomes` apparently. Could we do it better, how do you think?

Reply:
Yes, I think we should have a version of `assertTestFailed` that doesn't take a matcher, and delegate to it.